examples/swapAndBridge.ts

```diff
@@ -0,0 +1,182 @@
+const GNOSIS_CHAIN_BRIDGE = '0x88ad09518695c6c3712AC10a214bE5109a655671';
+
+const swapAndBridge: Parameters<typeof withAnvilProvider>[0] = async (
+  provider: ethers.JsonRpcProvider,
+  signers: ethers.Wallet[],
+  { factory, implementation, initCode }
+) => {
+  const user = signers[1];
+  const userAddr = await user.getAddress();
+
+  // wrap the ether and approve to cowswap
+  const amt = ethers.parseEther('1');
+  const wrapTx = await wrapEther(provider, userAddr, amt);
+  console.log('ether wrapped', wrapTx?.hash);
+  const approveTx = await approveToken(
+    provider,
+    WETH,
+    userAddr,
+    VAULT_RELAYER,
+    amt
+  );
+  console.log('weth approved', approveTx?.hash);
+
+  // compute the proxy address with CowShedSdk
+  const shedSdk = new CowShedSdk({
+    factoryAddress: factory,
+    implementationAddress: implementation,
+    chainId: 1,
+    proxyCreationCode: initCode,
+  });
+  const proxyAddress = shedSdk.computeProxyAddress(userAddr);
+  console.log('Proxy address for user', userAddr, 'is', proxyAddress);
+
+  const buyAmount = 10_000000n;
+  const validTo = Math.floor(new Date().getTime() / 1000) + 7200;
+
+  const order: Order = {
+    sellToken: WETH,
+    buyToken: USDC,
+    receiver: proxyAddress,
+    sellAmount: amt,
+    buyAmount,
+    validTo,
+    appData: '',
+    feeAmount: 0n,
+    kind: OrderKind.SELL,
+    partiallyFillable: true,
+    sellTokenBalance: OrderBalance.ERC20,
+    buyTokenBalance: OrderBalance.ERC20,
+  };
+
+  // post hooks
+  const calls: ICall[] = [
+    // approve the bridge to spend the swapped usdc
+    {
+      target: USDC,
+      callData: fnCalldata(
+        'approve(address,uint256)',
+        ABI_CODER.encode(['address', 'uint'], [GNOSIS_CHAIN_BRIDGE, buyAmount])
+      ),
+      value: 0n,
+      isDelegateCall: false,
+      allowFailure: false,
+    },
+    // bridge the usdc
+    {
+      target: GNOSIS_CHAIN_BRIDGE,
+      callData: fnCalldata(
+        'relayTokens(address,address,uint256)',
+        ABI_CODER.encode(
+          ['address', 'address', 'uint'],
+          [USDC, userAddr, buyAmount]
+        )
+      ),
+      value: 0n,
+      isDelegateCall: false,
+      allowFailure: false,
+    },
+  ];
+  const nonce = ethers.encodeBytes32String('first');
+
+  // signing the hooks intent
+  const hashToSign = shedSdk.hashToSignWithUser(
+    calls,
+    nonce,
+    BigInt(validTo),
+    userAddr
+  );
+  console.log('hash to sign', hashToSign);
+  const signature = user.signingKey.sign(hashToSign);
+  console.log('actual signature', signature.r, signature.s, signature.v);
+  const encodedSignature = CowShedSdk.encodeEOASignature(
+    BigInt(signature.r),
+    BigInt(signature.s),
+    signature.v
+  );
+
+  const hooksCalldata = CowShedSdk.encodeExecuteHooksForFactory(
+    calls,
+    nonce,
+    BigInt(validTo),
+    userAddr,
+    encodedSignature
+  );
+  const hooks = {
+    post: [
+      { target: factory, callData: hooksCalldata, gasLimit: '1000000000' },
```

Comment:
I think the example should methodology for determining a sane `gasLimit` (note that given the use of `ecrecover` within the signature verification, the gas used isn't fixed)

Reply:
added `provider.estimateGas` but its gonna fail for the prehooks usually since those tokens wont exist on the proxy at that point.